### PR TITLE
+str,doc #23274 add ASCII diagrams for Flow.fromSinkAndSource

### PR DESCRIPTION
**Pull request by ktoso**
That was pretty fun so I took a shot at it right away.
Resolves https://github.com/akka/akka/issues/23274 perhaps, in a way, somehow

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 1]
```diff
@@ -170,35 +223,37 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
...
-   *     | Resulting Flow             |
-   *     |                            |
-   *     |  +------+        +------+  |
-   *     |  |      |        |      |  |
-   * In ~~> | this | ~Out~> | flow | ~~> T
-   *     |  |      |        |      |  |
-   *     |  +------+        +------+  |
-   *     +----------------------------+
+   *     +---------------------------------+
+   *     | Resulting Flow[In, T, Mat]  |
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Looks like this line should be a bit wider

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 2]
```diff
@@ -474,19 +531,21 @@ trait FlowOps[+Out, +Mat] {
...
-   *     | Resulting Flow             |
-   *     |                            |
-   *     |  +------+        +------+  |
-   *     |  |      |        |      |  |
-   * In ~~> | this | ~Out~> | flow | ~~> T
-   *     |  |      |        |      |  |
-   *     |  +------+        +------+  |
-   *     +----------------------------+
+   *     +---------------------------------+
+   *     | Resulting Flow[In, T, Mat]  |
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: (and here)

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 3]
```diff
@@ -316,17 +318,45 @@ object Flow {
...
   def fromSinkAndSource[I, O](sink: Graph[SinkShape[I], _], source: Graph[SourceShape[O], _]): Flow[I, O, NotUsed] =
     fromSinkAndSourceMat(sink, source)(Keep.none)
 
   /**
    * Creates a `Flow` from a `Sink` and a `Source` where the Flow's input
    * will be sent to the Sink and the Flow's output will come from the Source.
+   * 
+   * The resulting flow can be visualized as:
+   * {{{
+   *     +-------------------------------------------------------+
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: This line seems to be a character too wide

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 4]
```diff
@@ -70,6 +83,19 @@ object Flow {
    * Creates a `Flow` from a `Sink` and a `Source` where the Flow's input
    * will be sent to the Sink and the Flow's output will come from the Source.
    *
+   * The resulting flow can be visualized as:
+   * {{{
+   *     +-------------------------------------------------------+
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Also 1 character too wide

**Review by raboof — COMMENTED** [review]
Very very nice! A couple of alignment issues (and scalariform complains about whitespace).

**Review by ktoso — COMMENTED** [review]
Alignments fixed

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 5]
```diff
@@ -21,8 +21,6 @@ import scala.compat.java8.FutureConverters._
 
 object Flow {
 
-  private[this] val _identity = new javadsl.Flow(scaladsl.Flow[Any])
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Unused?

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 5 above)
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Yeah, likely remained here from some ancient version where we added identity in places

**Review by raboof — APPROVED** [review]
Now it's just scalariform/jenkins complaining about whitespace ;)

**Comment by ktoso** [comment]
Each time a PR fails on whitespace I feel like it's a "salmon day" ( https://www.youtube.com/watch?v=GE94BJg3U1Q ) 😜  

Fixed.

**Comment by ktoso** [comment]
Validation OK, LGTMs gathered, merging

**Pull request merged**